torch/_decomp/decompositions.py

```diff
@@ -847,6 +847,70 @@ def native_group_norm(
     return (out, mean, rstd)
 
 
+@register_decomposition(aten.native_group_norm_backward)
+def native_group_norm_backward(
+    grad_output: Tensor,
+    input: Tensor,
+    mean: Tensor,
+    rstd: Tensor,
+    gamma: Optional[Tensor],
+    N: int,
+    C: int,
+    HxW: int,
+    group: int,
+    output_mask: List[bool],
+) -> Tuple[Optional[Tensor], Optional[Tensor], Optional[Tensor]]:
+    assert grad_output.shape == input.shape
+    assert input.numel() == N * C * HxW
+    assert mean.shape == (N, group)
+    assert rstd.shape == (N, group)
+    assert gamma is None or gamma.numel() == C
+
+    # Compute Internal gradients
+    acc_dtype = utils.get_computation_dtype(input.dtype)
```

Comment:
probably do @pw_cast_for_opmath instead of this, to make sure all inputs are converted

Reply:
wait is `pw_cast_for_opmath` the right decorator to use here? Do you want to upcast all inputs to fp32?

Reply:
idk how it's implemented in eager, but morally yes, all ops should be in fp32? E.g. in batch norm backward all intermediate computations are in fp32. 
We probably should be able to check that all inputs are of expected type, and type promotion decorations prevent us from doing so, but oh well.